raft/rafttest/interaction_env_handler_process_append_thread.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafttest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/datadriven"
+	"go.etcd.io/etcd/raft/v3"
+	"go.etcd.io/etcd/raft/v3/raftpb"
+)
+
+func (env *InteractionEnv) handleProcessAppendThread(t *testing.T, d datadriven.TestData) error {
+	idxs := nodeIdxs(t, d)
+	for _, idx := range idxs {
+		var err error
+		if len(idxs) > 1 {
+			fmt.Fprintf(env.Output, "> %d processing append thread\n", idx+1)
+			env.withIndent(func() { err = env.ProcessAppendThread(idx) })
+		} else {
+			err = env.ProcessAppendThread(idx)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ProcessAppendThread runs processes a single message on the "append" thread of
+// the node with the given index.
+func (env *InteractionEnv) ProcessAppendThread(idx int) error {
+	n := &env.Nodes[idx]
+	if len(n.AppendWork) == 0 {
+		return fmt.Errorf("no append work to perform")
```

Comment:
Is this really an error?

Reply:
Note: if there is no any parameter, please use errors.New instead.

Reply:
Done.